### ISSUE TITLE
Generated certificate does not match the hostname

### ISSUE THREAD

**Issue by kurtismullins** [quay/quay-operator#176]
**Description**

When deploying Quay with the latest code in master to GKE with the following configuration:
```
apiVersion: redhatcop.redhat.io/v1alpha1
kind: QuayEcosystem
metadata:
  name: example-quayecosystem
spec:
  quay:
    image: quay.io/projectquay/quay:latest
    imagePullSecretName: redhat-quay-pull-secret
    keepConfigDeployment: true
    externalAccess:
      hostname: example.com
      type: LoadBalancer
```
The hostname is forced to the IP Address of the `Service`. Attempting to override it as demonstrated in the documentation does not work.

This blocks the ability to use the Configuration Application because it attempts to validate the self-signed certificate and finds that it does not match the hostname.

**Steps to Reproduce**

- Deploy a Quay Instance. Keep the Config app running.
- Access the config app.
- Don't make any changes. Just click "Save Configuration".
- The Configuration App will complain that the certificate does not match the hostname.

**Acceptance Criteria**

- The Quay Config application should not fail when validating the generated certificate.
- The CR should be able to override the hostname.
- If a hostname is automatically assigned, such as a subdomain or IP address, and no certificate is supplied, then I would assume the self-signed certificate to be generated for the correct host. (Debatable)

**Comment by kurtismullins** [comment]
I believe one solution for handling the self-signed certificate is to verify that an existing certificate matches the current `hostname`.

It appears that [this conditional](https://github.com/redhat-cop/quay-operator/blob/62dc01ad2ce421f794e546e1cdf3c8283bab454c/pkg/controller/quayecosystem/provisioning/provision.go#L811) is where all of the magic happens.

**Comment by kurtismullins** [comment]
After some additional testing, it appears that this does not happen when using `routes` on OpenShift.

**Comment by kurtismullins** [comment]
This appears fixed in the 1.1.0 release candidate. Closing.
